src/mlpack/core/data/dictionary_encoding.hpp

```diff
@@ -0,0 +1,154 @@
+/**
+ * @file dictionary_encoding.hpp
+ * @author Jeffin Sam
+ *
+ * Definition of dictionary encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_DICTIONARY_ENCODING_HPP
+#define MLPACK_CORE_DATA_DICTIONARY_ENCODING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+#include <utility>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple Dictionary Enocding class
```

Comment:
Could you add a couple of lines to the comment?

Reply:
Sure.